src/bake/DevServer.zig

```diff
@@ -2738,6 +2867,85 @@ pub fn handleParseTaskFailure(
     }
 }
 
+/// We rely a lot on the files we parse in the bundle graph to know which files
+/// to add to the watcher.
+///
+/// There is one wrinkle with this:
+///
+/// If an entrypoint is deleted it will never get parsed and then never will be
+/// watched
+///
+/// So if we get `error.FileNotFound` on an entrypoint, we'll manually add it to
+/// the watcher to pick up if it got changed again.
+fn handleEntrypointNotFoundIfNeeded(dev: *DevServer, abs_path: []const u8, graph_kind: bake.Graph, bv2: *BundleV2) void {
```

Comment:
this should probably be abstracted to the watcher itself? "sticky_files" that always get re-watched automatically?